janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -267,6 +267,16 @@
                 "It is recommended to always enable index store names cache unless you have more then 50000 indexes " +
                 "per index store.", ConfigOption.Type.MASKABLE, true);
 
+    public static final ConfigOption<Integer> CONNECT_TIMEOUT =
+        new ConfigOption<>(ELASTICSEARCH_NS, "connect-timeout",
+            "Sets the maximum connection timeout (in milliseconds).", ConfigOption.Type.MASKABLE,
+            Integer.class, 1000);
```

Comment:
Why 1000? The default value for Apache HTTP client seems to be -1.

Reply:
@li-boxuan thanks for the review. I'm not as familiar with the more recent ES updates but isn't the ES rest client being used as opposed to Apache HTTP? If so, I think these defaults line up with what is [documented](https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/_timeouts.html).

Reply:
Thanks, you are right. ES rest-client internally uses Apache HTTP client (we are using 4.x HTTP client) which [uses -1 as the default value ](https://github.com/apache/httpcomponents-client/blob/4.5.x/httpclient/src/main/java/org/apache/http/client/config/RequestConfig.java#L280-L292)(which is interpreted as an infinite timeout), but [ES overrides these values](https://github.com/elastic/elasticsearch/blob/master/client/rest/src/main/java/org/elasticsearch/client/RestClientBuilder.java#L50-L51). So the default values indeed are 1000, but shall we use `RestClientBuilder.DEFAULT_CONNECT_TIMEOUT_MILLIS`?

Reply:
Yes, agreed, changed to the constant values.